lib/BloqadeSchema/src/execute.jl

```diff
@@ -164,27 +166,18 @@ end
 
 
 function to_schema(h::BloqadeExpr.RydbergHamiltonian, params::SchemaTranslationParams)
-    atoms,ϕ,Ω,Δ,info = hardware_transform_parse(h,params.device_capabilities)
-
-    # extract Detuning mask
-    Δ = info.Δ_mask.Δ
-    δ = info.Δ_mask.δ
-    Δi = info.Δ_mask.Δi
-
-    @debug "Hardware transform report: after linear interpolation ∫dt |ϕ(t)-ϕ_hw(t)| = $(info.ϕ_error) rad⋅μs"
-    @debug "Hardware transform report: after linear interpolation ∫dt |Ω(t)-Ω_hw(t)| = $(info.Ω_error) rad"
-    @debug "Hardware transform report: after linear interpolation ∫dt |Δ(t)-Δ_hw(t)| = $(info.Δ_error) rad"
-    @debug "Hardware transform report: mean deviation after rounding positions $(info.mse_atoms) μm"
+    atoms,ϕ,Ω,Δ,δ,Δi = schema_parse_rydberg_fields(h)
 
     violations = validate_analog_params(atoms,ϕ,Ω,Δ,δ,Δi,params.device_capabilities)
 
     if length(violations) > 0
-        throw(ValidationException())
+        show(violations)
```

Comment:
this should be cleaned up?

Reply:
I thought we would throw an error after showing all the violations. That was the behavior we had discussed I thought.

Reply:
I can add a message in the error but I just want the users to look at the violations

Reply:
sry, I probably wasn't clear, I was thinking to return a violation report in the validate process. So there won't be error at all in this stage, but a validate report object. This object can be empty, then the next pass will take over, or the next pass will complain if the report is not empty.

Reply:
if we print the violations here, then there is no big difference then putting the list of violations into the exception object. But then this is a bit strange as validate function should be expected as a nothrow function (so things do not terminate in the middle, and it would collect all the violations and return back the corresponding location of violation).

also FYI. to print something, it is always preferred to use explicit IO object instead of `stdout` since this won't work well when the output stream is an editor or notebook.

Reply:
> also FYI. to print something, it is always preferred to use explicit IO object instead of `stdout` since this won't work well when the output stream is an editor or notebook.

good point, I will keep that in mind. 

> sry, I probably wasn't clear, I was thinking to return a violation report in the validate process. So there won't be error at all in this stage, but a validate report object. This object can be empty, then the next pass will take over, or the next pass will complain if the report is not empty.
> if we print the violations here, then there is no big difference then putting the list of violations into the exception object. But then this is a bit strange as validate function should be expected as a nothrow function (so things do not terminate in the middle, and it would collect all the violations and return back the corresponding location of violation).

`validate_analog_params ` is the no throw function it simply lists all the violations as we had discussed. The line pointed to here is inside `to_schema` which is the "next function" which is complains. I agree printing the errors isn't great maybe I can pass the `violations` object back to the user through the `ValidationException` object. If I overload the `show` method for `ValidationException` it will allow me to print out the violations in a clean way?

Reply:
@Roger-luo checkout this [commit](https://github.com/QuEraComputing/Bloqade.jl/pull/430/commits/ddf34ef48bae4880d14da09df72e251bb75f3581)

Reply:
yeah I think the solution in that commit is better if it's inside `to_schema` since we will have pretty printing from the error object that works in other type of IO too.